addons/namingng.py

```diff
@@ -28,78 +28,70 @@
 import argparse
 import json
 
+errors = []
 
 # Auxiliary class
 class DataStruct:
     def __init__(self, file, linenr, string):
         self.file = file
         self.linenr = linenr
         self.str = string
+        self.column = 0
 
-
-def reportError(filename, linenr, severity, msg):
-    message = "[{filename}:{linenr}] ( {severity} ) naming.py: {msg}\n".format(
-        filename=filename,
-        linenr=linenr,
-        severity=severity,
-        msg=msg
-    )
-    sys.stderr.write(message)
-    return message
-
+def reportNamingError(location,message,errorId='namingConvention',severity='style',extra=''):
+    line = cppcheckdata.reportError(location,severity,message,'namingng',errorId,extra)
+    errors.append(line)
 
 def loadConfig(configfile):
     with open(configfile) as fh:
         data = json.load(fh)
     return data
 
 
-def checkTrueRegex(data, expr, msg, errors):
+def checkTrueRegex(data, expr, msg):
     res = re.match(expr, data.str)
     if res:
-        errors.append(reportError(data.file, data.linenr, 'style', msg))
+        reportNamingError(data,msg)
 
 
-def checkFalseRegex(data, expr, msg, errors):
+def checkFalseRegex(data, expr, msg):
     res = re.match(expr, data.str)
     if not res:
-        errors.append(reportError(data.file, data.linenr, 'style', msg))
+        reportNamingError(data,msg)
 
 
-def evalExpr(conf, exp, mockToken, msgType, errors):
+def evalExpr(conf, exp, mockToken, msgType):
     if isinstance(conf, dict):
         if conf[exp][0]:
             msg = msgType + ' ' + mockToken.str + ' violates naming convention : ' + conf[exp][1]
-            checkTrueRegex(mockToken, exp, msg, errors)
+            checkTrueRegex(mockToken, exp, msg)
         elif ~conf[exp][0]:
             msg = msgType + ' ' + mockToken.str + ' violates naming convention : ' + conf[exp][1]
-            checkFalseRegex(mockToken, exp, msg, errors)
+            checkFalseRegex(mockToken, exp, msg)
         else:
             msg = msgType + ' ' + mockToken.str + ' violates naming convention : ' + conf[exp][0]
-            checkFalseRegex(mockToken, exp, msg, errors)
+            checkFalseRegex(mockToken, exp, msg)
     else:
         msg = msgType + ' ' + mockToken.str + ' violates naming convention'
-        checkFalseRegex(mockToken, exp, msg, errors)
+        checkFalseRegex(mockToken, exp, msg)
 
 
 def process(dumpfiles, configfile, debugprint=False):
-
-    errors = []
```

Comment:
I wonder is there a special reason to make errors global.

Reply:
Not really, but passing around `errors` all the way down to the point where errors are finally logged, felt a bit pointless in a script like this. I actually think collecting errors might also be removed from the code; the verification can be done in a different way, comparing the output as it is produced by the script, and then the only reason left to collect errors is to print the line "Found errors: N" - is that line important to keep?

Reply:
> I actually think collecting errors might also be removed from the code

Yes it's really clumpsy/ugly to hard code the expected test output in this script. so we should do something else. We have some tests that utilize the inline suppressions instead. Okay.. if our plan is to get rid of errors anyway I don't have to worry about this detail..

> and then the only reason left to collect errors is to print the line "Found errors: N" - is that line important to keep?

I don't feel that is important.